src/Adapter/Carrier/CommandHandler/AddCarrierHandler.php

```diff
@@ -56,6 +58,10 @@ public function __construct(
      */
     public function handle(AddCarrierCommand $command): CarrierId
     {
+        if (!$command->getShopConstraint()->isEqual(ShopConstraint::allShops())) {
+            throw new CarrierException('Cannot add new carrier if shop constraint is not allShops');
```

Comment:
I believe @boherm intriduce a dedicated exception or at least an error code But this can be homogenized in another PR